plugin-server/src/utils/db/db.ts

```diff
@@ -202,6 +211,14 @@ export class DB {
                 } else {
                     return await this.postgres.query(queryString, values)
                 }
+            } catch (error) {
+                if (
+                    error.message &&
+                    POSTGRES_UNAVAILABLE_ERROR_MESSAGES.some((message) => error.message.includes(message))
```

Comment:
Is there a less fragile way to do this check? It seems that if say node-pg updates their error messages then we'll silently miss the appropriate error. Maybe there are error codes we could use instead? Or `instanceof`, whatever that doesn.

Reply:
Unfortunately I dug into their source code and couldn't find anything to hook onto. Can dig more. It seems there's also errors being thrown in dependencies of theirs, making this hard. For instance, if you look at https://app.posthog.com/instance/dead_letter_queue you'll see an `OperationalError`. That doesn't come from node-postgres directly though.

Reply:
Oh, actually `OperationalError` was thrown in Django 👀  I'll keep digging